env0/resource_environment.go

```diff
@@ -653,3 +682,28 @@ func resourceEnvironmentImport(ctx context.Context, d *schema.ResourceData, meta
 		return []*schema.ResourceData{d}, nil
 	}
 }
+
+func waitForDeployment(d *schema.ResourceData, apiClient client.ApiClientInterface) error {
+	for {
+		env, err := apiClient.Environment(d.Id())
+		if err != nil {
+			return err
+		}
+		switch env.LatestDeploymentLog.Status {
+		case "IN_PROGRESS",
+			"QUEUED",
+			"WAITING_FOR_USER":
+			// Deployment still in progress - wait 1 second then check again
+			time.Sleep(1 * time.Second)
+
+			// Deployment succeeded
+		case "SUCCESS",
+			"SKIPPED":
+			return nil
+		default:
+			// All other statuses must be failures
+			return fmt.Errorf("failed deploying environment: %v",
+				env.LatestDeploymentLog.Status)
```

Comment:
I think it should be something like the following:
```suggestion
			return fmt.Errorf("Environment deployment reached failure status: %v",
				env.LatestDeploymentLog.Status)
```